src/GraphQL.ApiTests/BomTests.cs

```diff
@@ -0,0 +1,36 @@
+namespace GraphQL.ApiTests;
+
+public class BomTests
+{
+    // https://github.com/graphql-dotnet/graphql-dotnet/pull/3477
+    [Fact]
+    public void Files_Should_Not_Use_BOM()
+    {
+        var baseDir = new DirectoryInfo(AppDomain.CurrentDomain.BaseDirectory).Parent!.Parent!.Parent!.Parent!.Parent!;
```

Comment:
This kind of code makes for unreliable tests.  Perhaps it may build to a different folder with a different build script, or different operating system, or a new version of a SDK.  I can't approve this PR.

Besides, it is unnecessary.  "So what" if BOM is or is not there?

Reply:
I can change directory location algorithm in any appropriate way.

> "So what" if BOM is or is not there?

During the long time, I see that sometimes one or another file sometimes changes in PRs because of this. These are parasitic changes that should not be. It's like if we constantly change spaces count from PR to PR.

Reply:
OK to use `GITHUB_WORKSPACE` from here https://docs.github.com/en/actions/learn-github-actions/environment-variables as root folder (if not null)?

Reply:
`CallerFilePath` looks better